include/broker/telemetry/metric_registry_impl.hh

```diff
@@ -9,6 +9,37 @@
 
 namespace broker::telemetry {
 
+// Collector for the broker::telemetry layer. We string through the label
+// values of individual metric instances as there aren't direct accessors
+// for those from a hdl.
+class metrics_collector {
+public:
+  virtual void operator()(const dbl_counter_family_hdl* family,
+                          const dbl_counter_hdl* counter,
+                          const std::vector<std::string_view>& label_values)
```

Comment:
1. Having only the values here makes it very cumbersome to correlate them back to the keys. Basically forces users to go back to the `family` and rely on index lookups.
2. Forcing the consumer to care about `std::vector` here also seems unnecessary.

If you prefer not lifting CAF's `label` type, this should at least least become `label_list labels`, with:

```cpp
  /// A list of key-value pairs representing the label dimensions of a metric.
  using label_list = span<const std::pair<std::string_view, std::string_view>>;
```